pymongo/mongo_client.py

```diff
@@ -1978,11 +1979,13 @@ def __init__(self, client, server, session):
         # of the pool at the time the connection attempt was started."
         self.sock_generation = server.pool.generation
         self.completed_handshake = False
+        self.service_id = None
 
     def contribute_socket(self, sock_info):
         """Provide socket information to the error handler."""
         self.max_wire_version = sock_info.max_wire_version
         self.sock_generation = sock_info.generation
+        self.service_id = sock_info.service_id
```

Comment:
What is `service_id` when there is no LB?

Reply:
It's always `None` when not in LB mode. When in LB mode there is a single connection pool and service_id is used to identify connections to the same host (and differentiate connections to different hosts). Take a peak at https://github.com/mongodb/mongo-python-driver/pull/628  and https://github.com/mongodb/specifications/blob/master/source/load-balancers/load-balancers.rst#error-handling for more info.